subprocess/subtree_oom.go

```diff
@@ -0,0 +1,180 @@
+package subprocess
+
+import (
+	"os/exec"
+	"runtime"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type OOMTracker struct {
+	WasOOMKilled bool  `json:"was_oom_killed"`
+	IsSudo       bool  `json:"is_sudo"`
+	Pids         []int `json:"pids"`
+}
+
+func NewOOMTracker() *OOMTracker {
+	oomTracker := OOMTracker{}
+	return &oomTracker
+}
+
+func (o *OOMTracker) Clear() error {
+	return o.clear()
+}
+
+func (o *OOMTracker) Check() error {
+	return o.check()
+}
```

Comment:
why do we need to have private copies of these functions?

Reply:
@bsamek mentioned we should have wrappers to hide functionality. Did we mean for the endpoint part of this or?

Reply:
What Sam means is that the body of `o.check()` can actually just go in `Check()`. The implementation is still hidden from the caller (since all the caller knows is the function signature of public functions).